bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationHandler.java

```diff
@@ -0,0 +1,454 @@
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.heliosventilation.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TooManyListenersException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEvent;
+import org.eclipse.smarthome.io.transport.serial.SerialPortEventListener;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link HeliosVentilationHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class HeliosVentilationHandler extends BaseThingHandler implements SerialPortEventListener {
+    private static final int BUSMEMBER_MAINBOARD = 0x11;
+    private static final int BUSMEMBER_SLAVEBOARDS = 0x10;
+    private static final byte BUSMEMBER_CONTROLBOARDS = (byte) 0x20;
+    private static final int BUSMEMBER_REC_MASK = 0xF0; // interpreting frames delivered to BUSMEMBER_ME &
+                                                        // BUSMEMBER_REC_MASK
+    private static final int BUSMEMBER_ME = 0x2F; // used as sender when communicating with the helios system
+    private static final int POLL_OFFLINE_THRESHOLD = 3;
+
+    /** Logger Instance */
+    private final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private @NonNullByDefault({}) HeliosVentilationConfiguration config;
```

Comment:
There is a bug in the core that causes `handleCommand` to sometimes get called before the handler's `initialize` is called. So we recommend to just populate the configuration with a default instance to avoid NPE issues.
```suggestion
    private HeliosVentilationConfiguration config = new HeliosVentilationConfiguration();
```